src/cpp/transport/UDPv6Transport.cpp

```diff
@@ -225,35 +272,65 @@ bool UDPv6Transport::OpenInputChannel(const Locator_t& locator, TransportReceive
     {
         // The multicast group will be joined silently, because we do not
         // want to return another resource.
-        auto pChannelResource = mInputSockets.at(IPLocator::getPhysicalPort(locator));
-
-        std::vector<IPFinder::info_IP> locNames;
-        GetIP6sUniqueInterfaces(locNames);
-        for (const auto& infoIP : locNames)
+        auto pChannelResources = mInputSockets.at(IPLocator::getPhysicalPort(locator));
+        for (auto& pChannelResource : pChannelResources)
         {
-            auto ip = asio::ip::address_v6::from_string(infoIP.name);
-            try
-            {
-                pChannelResource->getSocket()->set_option(ip::multicast::join_group(
-                        ip::address_v6::from_string(IPLocator::toIPv6string(locator)), ip.scope_id()));
-            }
-            catch(std::system_error& ex)
+            std::vector<IPFinder::info_IP> locNames;
+            GetIP6sUniqueInterfaces(locNames);
+            for (const auto& infoIP : locNames)
             {
-                (void)ex;
-                logWarning(RTPS_MSG_OUT, "Error joining multicast group on " << ip << ": "<< ex.what());
+                auto ip = asio::ip::address_v6::from_string(infoIP.name);
+                if (IsInterfaceAllowed(ip))
+                {
+                    try
+                    {
+                        pChannelResource->getSocket()->set_option(ip::multicast::join_group(
+                            ip::address_v6::from_string(IPLocator::toIPv6string(locator)), ip.scope_id()));
+                    }
+                    catch (std::system_error& ex)
+                    {
+                        (void)ex;
+                        logWarning(RTPS_MSG_OUT, "Error joining multicast group on " << ip << ": " << ex.what());
+                    }
+                }
             }
         }
     }
 
     return success;
 }
 
-bool UDPv6Transport::IsInterfaceAllowed(const std::string& interface)
+std::vector<std::string> UDPv6Transport::GetInterfacesList(const Locator_t& locator)
+{
+    std::vector<std::string> vOutputInterfaces;
+    if (IsInterfaceWhiteListEmpty() || (!IPLocator::isAny(locator) && !IPLocator::isMulticast(locator)))
+    {
+        if (IPLocator::isMulticast(locator))
+        {
+            vOutputInterfaces.push_back("::1");
```

Comment:
Here you have to return the any address (::0)

Reply:
Fixed